### PR TITLE
added waiter for navbar.spinner element to menu_click()

### PR DESCRIPTION
**Pull request by rplevka**
added waiter for navbar.spinner element to menu_click()

### CODE REVIEW

**Comment by rplevka** [comment]
as discussed on our automation call, This should fix some issues caused by missed running ajax calls during the menu navigation.

This change implements a wait for navbar spinner element disappearance on menu_click() method.

Please note that the more appropriate place for the waiter would be directly in the `wait_for_ajax` method (to apply to all actions, not just `menu_click`) but it produces a lot of `Accessing locator navbar.spinner` DEBUG messages.

This also won't fix all timeout issues, as it has a hard timeout value set (30 secs). There are actions (e.g. Dashboard loading) which can easily extend this timeout (>30 secs).

**Comment by sthirugn** [comment]
can you please update the pr/commit description to be more relevant to the code changes?

**Comment by rochacbruno** [comment]
@rplevka I see you removed the 5900 port binding that was previously in your branch and as @elyezer mentioned that we need to check if there is a firewall blocking that port in order to work.

So are we going to keep the VNC connection as an option or just stick with screenshots?

**Comment by elyezer** [comment]
Please update the commit information.

ACK, pending commit information update.

**Comment by sthirugn** [comment]
Once the commit message is updated, I can merge and trigger auto

**Comment by rplevka** [comment]
@rochacbruno yes, I removed the VNC feature completely as it would fail the tests straight on `setUp`.
the only thing that remains is the `navbar.spinner` waiter with `bz_is_open` condition.

File: robottelo/ui/navigator.py [hunk 1]
```diff
@@ -20,7 +21,14 @@ def menu_click(self, top_menu_locator, sub_menu_locator,
                 "arguments[0].click();",
                 tertiary_element,
             )
-        self.browser.refresh()
+        # 1328627 - as the number of subs increases, dashboard page gets slower
```
Comment (oshtaier) [inline robottelo/ui/navigator.py]: Hmm, why do we have reference to dashboard page at all? It is not a default sat page since 6.2

**Comment by rplevka** [comment]
@elyezer , @sthirugn , @rochacbruno - I'm happy to update the commit message, however looking at the standalone automation results, it does not seem to resolve many issues.
We can try it though.

**Comment by oshtaier** [comment]
pending question

File: robottelo/ui/navigator.py (hunk 1 above)
Comment (oshtaier) [inline robottelo/ui/navigator.py]: Sorry, as far as I see we have some changes in GA :)

Comment (rplevka) [inline robottelo/ui/navigator.py]: yeah, i was just trying to find out the exact point where this has been introduced :)

File: robottelo/ui/navigator.py [hunk 2]
```diff
@@ -20,7 +21,14 @@ def menu_click(self, top_menu_locator, sub_menu_locator,
...
+        wait_timeout = 12
+        if bz_bug_is_open(1328627):
```
Comment (oshtaier) [inline robottelo/ui/navigator.py]: Does it fetch BZ information every menu click operation?

Comment (rplevka) [inline robottelo/ui/navigator.py]: Fair point - here's the answer:
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/decorators.py#L252
tl;dr - no, it uses a cache, so every fetched BZ is saved in the `_bugzilla` dict for later use,

**Comment by oshtaier** [comment]
Not so happy to have code like this in the most fundamental methods, but after few more checks, agreed to ACK

**Comment by abalakh** [comment]
Fourth ACK :)
